sentry_streams/sentry_streams/adapters/arroyo/translator.py

```diff
@@ -0,0 +1,301 @@
+        # accumulator in the flushed window
+        self.accs[first_acc_id] = KafkaAccumulator(self.acc)
+
+    def __maybe_flush(self, cur_time: float) -> None:
+
+        for i in range(len(self.windows)):
+            window = self.windows[i]
+
+            if cur_time >= self.window_close_times[i]:
+                self.__merge_and_flush(i)
+
+                # only shift a window if it was flushed
+                window = [(t + len(window)) % len(self.accs) for t in window]
+                self.windows[i] = window
+
+                self.window_close_times[i] += float(self.window_size)
+
+    def submit(self, message: Message[Union[FilteredPayload, TPayload]]) -> None:
+
+        assert not self.__closed
+
+        value = message.payload
+        if isinstance(value, FilteredPayload):
+            self.next_step.submit(cast(Message[TResult], message))
+            return
+
+        assert isinstance(value, RoutedValue)
+        if value.route != self.route:
+            self.next_step.submit(cast(Message[TResult], message))
+            return
+
+        cur_time = time.time() - self.start_time
+        acc_id = int((cur_time % self.largest_val) // self.window_slide)
+
+        self.__maybe_flush(cur_time)
+
+        self.accs[acc_id].add(message.value)
+
+    def poll(self) -> None:
+        assert not self.__closed
+
+        cur_time = time.time() - self.start_time
+        self.__maybe_flush(cur_time)
+
+        self.next_step.poll()
+
+    def close(self) -> None:
+        self.__closed = True
+
+    def terminate(self) -> None:
+        self.__closed = True
+        self.next_step.terminate()
+
+    def join(self, timeout: Optional[float] = None) -> None:
+        self.next_step.close()
+        self.next_step.join()
+
+
+def build_arroyo_windowed_reduce(
+    streams_window: Window[MeasurementUnit],
+    accumulator: Callable[[], Accumulator[Any, Any]],
+    next_step: ProcessingStrategy[Union[FilteredPayload, TPayload]],
+    route: Route,
+) -> ProcessingStrategy[Union[FilteredPayload, TPayload]]:
+
+    match streams_window:
+        case SlidingWindow(window_size, window_slide):
+            match (window_size, window_slide):
+                case (timedelta(), timedelta()):
+
+                    size = window_size.total_seconds()
+                    slide = window_slide.total_seconds()
+
+                    if slide == 0.0:
+                        raise ValueError(f"Window slide {slide} cannot be 0")
+
+                    if not (size).is_integer() or not (slide).is_integer():
+                        raise ValueError(
+                            "Currently only second precision is supported for window size and window slide"
+                        )
+
+                    return WindowedReduce(
+                        size,
+                        slide,
+                        accumulator,
+                        next_step,
+                        route,
+                    )
+
+                case _:
```

Comment:
i guess this can be expressed at type level -- mypy should yell if you _create_ a `SlidingWindow(123, 123)` (with integers instead of timedeltas)

so ig split up `MeasurementUnit`